### ISSUE TITLE
Instance dropdown on Login form does not support filtering as you type

### ISSUE THREAD

**Issue by pekempy** [laurencedawson/sync-for-lemmy#146]
**Describe the bug**  
When logging into a new account, the instance list isn't sorted alphabetically or by any visible metric. 
When typing an instance name to search e.g typing `beeh` it doesn't filter the list down to matching result.

**Steps to Reproduce**  
1. Open the account switcher 
2. Click `Add Account`
3. Open the Instance dropdown
4. Type on keyboard an instance

**Expected behavior**  
When typing an instance name e.g. `beeha` only `beehaw.org` should show to make it easier for users to find their instance

**Screenshots**  
![image](https://github.com/laurencedawson/sync-for-lemmy/assets/22874875/f6b179f2-d69b-4b9d-8c20-208f224228dc)

**Smartphone (please complete the following information):**
 - Device: Pixel 6 Pro
 - Android: Android 14 Beta 4
 - Sync version: v23.07.28-01:03
 - Sync variant: for Lemmy :D

**Additional context**  
Alphabetical sorting may be helpful, or if it's sorted by another metric such as number of registered users (popularity) - show that variable in the dropdown too?

**Comment by laurencedawson** [comment]
This was intentional to allow users to quickly pick one of the main instances but open to suggestion.

**Comment by pekempy** [comment]
Yeah, just from a user perspective, as I type it should filter to what I'm currently typing. 
I was thinking something like a creatable select ([example](https://mantine.dev/core/select/#creatable) - I know it's react it's just what i was thinking behaviour should be like) which filters as you type, but then you can create options yourself so users can add instances which aren't listed?
